commons-vfs2-zookeeper/src/main/java/org/apache/commons/vfs2/provider/zookeeper/ZkFileProvider.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.commons.vfs2.provider.zookeeper;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.vfs2.Capability;
+import org.apache.commons.vfs2.FileName;
+import org.apache.commons.vfs2.FileSystem;
+import org.apache.commons.vfs2.FileSystemException;
+import org.apache.commons.vfs2.FileSystemOptions;
+import org.apache.commons.vfs2.provider.AbstractOriginatingFileProvider;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.CuratorFrameworkFactory.Builder;
+
+public class ZkFileProvider extends AbstractOriginatingFileProvider {
+  static final Collection<Capability> CAPABILITIES =
+      Collections.unmodifiableCollection(Arrays.asList(Capability.GET_TYPE,
+          Capability.READ_CONTENT, Capability.URI, Capability.WRITE_CONTENT,
+          Capability.DIRECTORY_READ_CONTENT, Capability.LIST_CHILDREN));
+
+  public ZkFileProvider() {
+    super();
+  }
+
+  @Override
+  protected FileSystem doCreateFileSystem(FileName rootName,
+      FileSystemOptions fileSystemOptions) throws FileSystemException {
+    ZkFileSystemConfigBuilder configBuilder =
+        ZkFileSystemConfigBuilder.getInstance();
+
+    // did the caller provide a curator framework?
+    CuratorFramework framework =
+        configBuilder.getCuratorFramework(fileSystemOptions);
+    if (framework == null) {
+      String connectionString =
+          configBuilder.getZkConnectionSring(fileSystemOptions);
+      RetryPolicy retryPolicy = configBuilder.getRetryPolicy(fileSystemOptions);
+      String namespace = configBuilder.getZkNamespace(fileSystemOptions);
+      int connectionTimeout =
```

Comment:
Please set your line length to 120 and reformat all; see `/commons-vfs2-project/checkstyle.properties`

Reply:
sorry, the check style file has been secretly failing for me in idea
```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: LineLength is not allowed as a child in Checker
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:423)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:138)
	at org.infernus.idea.checkstyle.service.cmd.OpCreateChecker.execute(OpCreateChecker.java:61)
	at
```

Reply:
I can't get the check style configuration to load, and even with the line lengths set to 120, I can't get it to reformat the lines _longer_ if that is what you are after.
If you have an ideas, I'm all ears.

Reply:
Use Eclipse ;-)

Reply:
But don't worry about the line length if it is too much of a headache, I can reformat after I bring the patch in.